uint32 rst
---
uint32 ret
